File: Server/Core/DTOs/NewMiracleDto.cs
using System.ComponentModel.DataAnnotations;

namespace Core.DTOs;

public class NewMiracleDto
{
    [RegularExpression(@"^(?=.*[A-Za-z]).+$", ErrorMessage = "Title cannot be only numbers.")]
    public string Title { get; set; } = default!;
    public string Country { get; set; } = default!;
    public int Century { get; set; }
    public string Image { get; set; } = default!;
    public string Description { get; set; } = default!;
    public string MarkdownContent { get; set; } = default!;

    public string? Date { get; set; }
    public string? LocationDetails { get; set; }
    public List<int> TagIds { get; set; } = [];
}

File: Client/src/app/features/miracles/interfaces/new-miracle-dto.ts
export interface NewMiracleDto {
  title: string;
  country: string;
  century: number;
  image: string;
  description: string;
  markdownContent: string;

  date?: string | null;
  location?: string | null;
  saintId?: number | null;
  tagIds: number[];
}
